lms/services/oauth_http.py

```diff
@@ -0,0 +1,49 @@
+class OAuthHTTPService:
+    """Send OAuth 2.0 requests and return the responses."""
+
+    def __init__(self, http_service, oauth2_token_service):
+        self._http_service = http_service
+        self._oauth2_token_service = oauth2_token_service
+
+    def get(self, *args, **kwargs):
+        return self.request("GET", *args, **kwargs)
+
+    def put(self, *args, **kwargs):
+        return self.request("PUT", *args, **kwargs)
+
+    def post(self, *args, **kwargs):
+        return self.request("POST", *args, **kwargs)
+
+    def patch(self, *args, **kwargs):
+        return self.request("PATCH", *args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        return self.request("DELETE", *args, **kwargs)
```

Comment:
:sob: - I think my general view on this is well known.

Another tack to take if we don't want to _be_ a session is to stop pretending to be one. Just make callers use `request` instead.

How many of these methods get called? Do we need these yet / at all?

Reply:
Originally it only had `request()` and Marcos requested that `get()` and pals be added for convenience. I don't think it's too bad: our services don't want to _be_ a session they just want to forward to _some_ of `requests.Sessions`'s methods (several of them), not everything. It means some duplicated boilerplate but they're just trivial forwarding methods